crates/core/src/api/method.rs

```diff
@@ -0,0 +1,59 @@
+use std::fmt::{self, Display};
+
+use revision::revisioned;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+	err::Error,
+	sql::{statements::info::InfoStructure, Value},
+};
+
+#[revisioned(revision = 1)]
+#[derive(Clone, Copy, Debug, Eq, PartialEq, PartialOrd, Serialize, Deserialize, Hash)]
+#[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
+#[non_exhaustive]
+pub enum Method {
+	Delete,
+	Get,
+	Patch,
+	Post,
+	Put,
+	Trace,
+}
+
+impl TryFrom<&Value> for Method {
+	type Error = Error;
+	fn try_from(value: &Value) -> Result<Self, Self::Error> {
+		match value {
+			Value::Strand(s) => match s.as_str() {
+				v if v.eq_ignore_ascii_case("delete") => Ok(Self::Delete),
+				v if v.eq_ignore_ascii_case("get") => Ok(Self::Get),
+				v if v.eq_ignore_ascii_case("patch") => Ok(Self::Patch),
+				v if v.eq_ignore_ascii_case("post") => Ok(Self::Post),
+				v if v.eq_ignore_ascii_case("put") => Ok(Self::Put),
+				v if v.eq_ignore_ascii_case("trace") => Ok(Self::Trace),
+				_ => Err(Error::Thrown("method does not match".into())),
+			},
```

Comment:
We could probably better match with s.to_lowercase() , so we don't have a per condition match

```rust
Value::Strand(s) => match s.to_lowercase() {
	"delete" => Ok(Self::Delete),
				...
```

Reply:
Should probably use `to_ascii_lowercase()`, for performance reasons?